src/app/pages/Register/Register.tsx

```diff
@@ -4,15 +4,32 @@ import BackButton from '../../components/BackButton/BackButton';
 import Logo from '../../components/Icons/Logo';
 import LabeledInput from '../../components/LabeledInput/LabeledInput';
 import Button from '../../components/Button/Button';
+import { User } from '../../../types';
+import { postUser } from '../../utils/api';
+import { useHistory } from 'react-router-dom';
 
 export default function Register(): JSX.Element {
+  const history = useHistory();
   const [userName, setUserName] = useState('');
   const [email, setEmail] = useState('');
```

Comment:
is can see that you are on a really good way 👍 

maybe you can think about adding a regEx pattern for your email and password validation :) 

for e.g. like this : https://www.tutorialspoint.com/regex-in-reactjs

but try to understand, what is happening there ->  https://regexr.com/